clients/cloud/rust/src/consumer.rs

```diff
@@ -0,0 +1,57 @@
+extern crate futures;
+extern crate rdkafka;
+extern crate tokio;
+
+use futures::stream::Stream;
+use rdkafka::consumer::stream_consumer::StreamConsumer;
+use rdkafka::consumer::Consumer;
+use rdkafka::Message;
+use std::boxed::Box;
+use tokio::runtime::current_thread::Runtime;
+
+mod utils;
+
+fn echo_message<M: Message>(msg: M) -> Result<(), std::str::Utf8Error> {
+    let deserialize = |o| match o {
+        None => Ok("".to_string()),
+        Some(val) => Ok(std::str::from_utf8(val)?.to_string()),
+    };
+
+    println!(
+        "message received from topic {} and partition {} with offset {}, key {}, value {}",
+        msg.topic(),
+        msg.partition(),
+        msg.offset(),
+        deserialize(msg.key())?,
+        deserialize(msg.payload())?,
+    );
+
+    Ok(())
+}
+
+fn main() -> Result<(), Box<std::error::Error>> {
+    let (topic, config) = utils::get_config()?;
+    let consumer: StreamConsumer = config.create()?;
+
+    consumer.subscribe(&vec![topic.as_ref()])?;
+
+    let processor = consumer
+        .start()
+        .filter_map(|result| match result {
+            Ok(_) => result.ok(),
+            Err(err) => {
+                eprintln!("error consuming from message stream: {}", err);
+                None
+            }
+        })
+        .for_each(move |msg| {
```

Comment:
nope :) good catch - I wish the compiler told you this because at one point I probably did but then when I no longer needed it it's still valid code and doesn't warn.